nsgapi.py

```diff
@@ -107,3 +110,72 @@ def make_nsgql_query_request(self, nsgql):
             }
         )
         return query
+
+    def get_device_tags(self, device_filter: dict = {}) -> dict:
+        """
+        :param device_filter: dict describes device match:
+                              one of "id": device_id (int)
+                              "name": device name (str)
+                              "address": device primary ip address
+        :return:              list of devices with list of device tags
+        """
+
+        url = self.concatenate_url('v2/tags/net/{0}/external/'.format(self.netid))
+        headers = self.make_headers()
+
+        resp = self.sess.get(url=url,
+                             headers=headers,
+                             params={k:v for k, v in device_filter if k in DEVICE_FIELDS_MATCH} or None,
+                             verify=False
+                             )
+        if resp.status_code != http.HTTPStatus.OK:
+            self.log.error('NetSpyGlass GET tags error: {} {0}'.format(resp.status_code, resp.text))
+            return list()
+
+        try:
+            result = resp.json()
+        except json.JSONDecodeError as e:
+            raise ValueError(e.msg)
+        return result
+
+    def post_device_tags(self, tag_list: list[dict]) -> dict:
+        """
+        :param tag_list: list of dict
+                            {
+                             "category": "device",
+                             "device": {} - dict to match device by "id": id or "address": address
+                             "tags": [] list of tags in format: tagName.tagValue
+                             }
+        :return:              list of devices with list of device tags
+        """
+
+        url = self.concatenate_url('v2/tags/net/{0}/external/'.format(self.netid))
+        headers = self.make_headers()
+        resp = self.sess.post(url=url,
+                              headers=headers,
+                              verify=False,
+                              json=tag_list
+                              )
+        if resp.status_code != http.HTTPStatus.OK:
+            self.log.error('NetSpyGlass POST tags error: {} {0}'.format(resp.status_code, resp.text))
+            return list()
+        return self.parse_and_log_response("ADD TAGS", resp)
+
+    def get_remote_config(self, asset_type: str ) -> str:
+        """
+        Download config file from gitea
+        :param asset_type: config file name
+        :return:
+        """
+        url = self.concatenate_url('store/ex_tags/{}'.format(asset_type))
```

Comment:
looks like it is possible to store configuration that describes mapping between Netbox attributes and NSG external tags in NSG store server. How would the user manage this mapping when it is stored there? We do not have the UI for it (and are unlikely to ever get resources to build it).

Reply:
it is a file in gitea 
for example on labqa01: https://labqa01.netspyglass.com:13148/nsgapps/labqa01/netspyglass/src/branch/master/etc/resources/external_tags/config.yaml

Reply:
I do not think we need this feature. I prefer when there is one way to do something, it is easier for the user and easier for us since we do not need to support multiple. The script takes config file that the user manages, this is a standard design, it is simple, easy to remember, and perfectly "good enough"

Reply:
removed

Reply:
you say you removed it but the function `get_remote_config()` is still there. Please remove it